### PR TITLE
Move edxapp vars to app

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

This is the second step of our application driven migration, now targeting the `edxapp` application

## Proposal

- [x] fix app/services labels in templates
- [x] define an host+port for the patient0/development project
- [x] move `edxapp` variables from `group_vars/all/main.yml` to `apps/edxapp/vars/all/main.yml` and use them in templates
- [x] all synchronous sequential job execution :tada: 
- [x] test `edxapp`'s deployment in the CI
- [x] wait for all services to be up before running jobs
- [x] ~add support for multiple hosts per-service~ **edit**: postponed to another PR (see #90)
- [x] ~add lms host definition and test it~ **edit**: postponed to another PR (see #90)

### CODE REVIEW

File: apps/edxapp/templates/lms/svc.yml.j2 [hunk 1]
```diff
@@ -2,17 +2,19 @@ apiVersion: v1
 kind: Service
 metadata:
   labels:
-    app: edxapp-lms
-    version: "{{ edxapp_tag }}"
+    app: lms
```
Comment (sampaccoud) [inline apps/edxapp/templates/lms/svc.yml.j2]: should be edxapp?

File: bin/dev [hunk 2]
```diff
@@ -2,88 +2,42 @@
 #
 # Arnold's dev script
 #
-# We use KVM as the default hypervisor, if you want to use VirtualBox instead,
-# please set the USE_KVM environment variable to 0, e.g.:
+# Berore running this script, please make sure that your system met the
```
Comment (sampaccoud) [inline bin/dev]: s/Berore/Before

File: bin/dev [hunk 3]
```diff
@@ -2,88 +2,42 @@
...
 #
-# USE_KVM=0 bin/dev
+#   1. Make sure that the "/etc/docker/daemon.json" file contains the
+#      "172.30.0.0/16" insecure registry:
+#
+#          {
+#              "insecure-registries" : ["172.30.0.0/16"]
+#          }
+#
+#    2. To run ES, you will to ensure that your kernel's vm.max_map_count
```
Comment (sampaccoud) [inline bin/dev]: you will NEED to ensure

File: bin/dev [hunk 4]
```diff
@@ -2,88 +2,42 @@
...
+function _start_cluster() {
 
+    OPENSHIFT_DOMAIN=$(hostname -I | awk '{print $1}')
+    K8S_AUTH_HOST="https://${OPENSHIFT_DOMAIN}:8443"
 
-# _login: login to minishift
-function _login() {
-    _set_minishift_path
-    oc login "https://$(minishift ip):8443" --username=developer --password=developer
+    oc cluster up --server-loglevel=5 --public-hostname="${OPENSHIFT_DOMAIN}"
```
Comment (sampaccoud) [inline bin/dev:33]: that's a bit more than login!?

File: group_vars/all/main.yml [hunk 5]
```diff
@@ -32,8 +32,8 @@ trashable_env_types:
 ## /!\  Variables below will soon be deprecated /!\ __________________________##
 
 # Ports
-cms_port:       8081
-lms_port:       8071
+# cms_port:       8081
+# lms_port:       8071
```
Comment (sampaccoud) [inline group_vars/all/main.yml]: dead code

File: group_vars/all/main.yml [hunk 6]
```diff
@@ -43,20 +43,20 @@ nginx_ports:
...
-mongodb_image:       "centos/mongodb-32-centos7"
-mongodb_tag:         "3.2"
-mysql_image:         "centos/mysql-57-centos7"
-mysql_tag:           "5.7"
+# edxapp_image:        "fundocker/edxapp"
+# edxapp_tag:          "ginkgo.1-1.0.3"
+# mongodb_image:       "centos/mongodb-32-centos7"
+# mongodb_tag:         "3.2"
+# mysql_image:         "centos/mysql-57-centos7"
+# mysql_tag:           "5.7"
```
Comment (sampaccoud) [inline group_vars/all/main.yml]: dead code

File: group_vars/all/main.yml [hunk 7]
```diff
@@ -85,10 +85,10 @@ lms_host: "{{ project_name }}-lms-{{ deployment_stamp }}.{{ domain_name }}"
...
-  port: "27017"
-mysql:
-  host: "mysql"
-  port: "3306"
+# mongodb:
+#   host: "mongodb"
+#   port: "27017"
+# mysql:
+#   host: "mysql"
+#   port: "3306"
```
Comment (sampaccoud) [inline group_vars/all/main.yml]: dead code

File: bin/dev (hunk 4 above)
Comment (jmaupetit) [reply bin/dev:33]: This function is no longer called login. Forget about this crappy diff.

File: apps/edxapp/templates/lms/svc.yml.j2 (hunk 1 above)
Comment (jmaupetit) [reply apps/edxapp/templates/lms/svc.yml.j2]: Wow, well spotted!

**Comment by jmaupetit** [comment]
We still have a timeout from CircleCI's test-bootstrap job. We decided to merge this as-is and address this timeout issue in a new PR: the idea is to split the test-bootstrap job in a per-app job, to hopefully prevent job failure due to a timeout issue.

**Pull request merged**